src/main/java/dev/jbang/Script.java

```diff
@@ -112,11 +115,13 @@ public Script(String script, List<String> arguments, Map<String, String> propert
 			p.putAll(properties);
 		}
 
-		List<String> dependencies = getLines()	.stream()
+		Stream<String> depStream = getLines()	.stream()
 												.filter(it -> isDependDeclare(it))
 												.flatMap(it -> extractDependencies(it))
-												.map(it -> PropertiesValueResolver.replaceProperties(it, p))
-												.collect(Collectors.toList());
+												.map(it -> PropertiesValueResolver.replaceProperties(it, p));
+
+		List<String> dependencies = Stream	.concat(depStream, additionalDeps.stream())
```

Comment:
this would make the additional deps get added *last*, right ? 

I would think that they should be first to enable overriding/affecting sequence of loading of resources.

Reply:
Aha, yes of course, in my head the later ones would override the earlier ones, but in a class path that doesn't make sense.